### PR TITLE
Limit short_vec length to u16, usize is overkill for our usage

### LINKED ISSUE #4586: short_vec doesn't validate the vector length on deserialization
shortvec doesn't validate the length on deserialization for sanity, which can cause the host process to OOM if it's passed malformed data:
https://github.com/solana-labs/solana/blob/1c765124e7c739326d77f44b09d6c5977374c35e/sdk/src/short_vec.rs#L125-L130

### PR DESCRIPTION
**Pull request by mvines**
Work around the data validation error at #4586 by restricting short_vec length to u16.

Fixes #4586

### CODE REVIEW

File: sdk/src/short_vec.rs [hunk 1]
```diff
@@ -172,42 +98,39 @@ impl<'de, T: Deserialize<'de>> Deserialize<'de> for ShortVec<T> {
...
-    bincode::serialize(&ShortUsize(len)).unwrap()
-}
-
 /// Return the decoded value and how many bytes it consumed.
 pub fn decode_len(bytes: &[u8]) -> (usize, usize) {
-    let short_len: ShortUsize = bincode::deserialize(bytes).unwrap();
-    let num_bytes = bincode::serialized_size(&short_len).unwrap() as usize;
-    (short_len.0, num_bytes)
+    let short_len: u8 = bincode::deserialize(bytes).unwrap();
+    let num_bytes = bincode::serialized_size(&short_len).unwrap() as u16;
```
Comment (carllin) [inline sdk/src/short_vec.rs]: why does this have to be a u16?

File: sdk/src/short_vec.rs [hunk 2]
```diff
@@ -149,7 +75,7 @@ where
     T: Deserialize<'de>,
 {
     let visitor = ShortVecVisitor { _t: PhantomData };
-    deserializer.deserialize_tuple(std::usize::MAX, visitor)
+    deserializer.deserialize_tuple(std::u16::MAX as usize, visitor)
```
Comment (carllin) [inline sdk/src/short_vec.rs]: hmm, why is this u16::MAX and not u8::MAX?

**Comment by mvines** [comment]
Ack, turns out we need u16.  The Message data can be > 255

**Comment by mvines** [comment]
s/u8/16/g

**Pull request merged**